### PR TITLE
Implement `rank_user_replies` in web

### LINKED ISSUE #33: Implement `rank_user_replies` in web
The `rank_user_replies` task type needs to display these fields:
*  `conversation.messages`: An messages in a conversation
*  `replies`: An array of replies to the conversation

It takes the following interaction type:
*  `type`: `post_ranking`
*  `ranking`: An array of ints representing the preferred ordering of the `replies`, indexed by zero.  example: `[2,1,0,3]` for a series of 4 replies.

### PR DESCRIPTION
**Pull request by AbdBarho**
Closes #33

Also started moving duplicate code into their own components

### CODE REVIEW

**Comment by fozziethebeat** [comment]
I see you marked this as draft.  Will approve once you set it to ready for review again.

**Comment by AbdBarho** [comment]
> I see you marked this as draft. Will approve once you set it to ready for review again.

I don't want to unset draft because I am not able to test against our backend.

Sampling random task of any kinds returns the error `No Threads Found`, 
I think the problem is that in case there are no threads, no new dummy ones are generated...

I will see if I can fix this

**Comment by yk** [comment]
this will be fixed by #144 , sorry for the inconvenience

File: website/src/components/Sortable/Sortable.tsx [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+            [newRanking[i], newRanking[newIdx]] = [newRanking[newIdx], newRanking[i]];
+            update(newRanking);
+          }}
+        >
+          {items[rank]}
+        </SortableItem>
+      ))}
+    </ul>
+  );
+};
```
Comment (k-nearest-neighbor) [inline website/src/components/Sortable/Sortable.tsx:48]: Looking to merge this in just to get the template and the bulk of the work so far in to the repo 🥳
But I think we should be able to replace with https://dndkit.com/ for the MVP

**Review by k-nearest-neighbor — APPROVED** [review]
Excellent. Ok I think there are 2 areas still to do with rank_user_replies:

* Drag and drop with https://dndkit.com/ 
* Some adjustments to the task description and possibly design; As a user I do not understand this task (what are the replies replying to?)

Approving this to merge in the progress

**Review by andreaskoepf — APPROVED** [review]
trusting knn. :-)

**Pull request merged**